### ISSUE TITLE
Enabling progress bar causes shell script to use remove mode

### ISSUE THREAD

**Issue by phemmer** [sahib/rmlint#270]
```
# mkdir /tmp/test

# echo hi > /tmp/test/a

# echo hi > /tmp/test/b

# rmlint -o summary -T duplicates -c sh:hardlink -o sh:/tmp/out /tmp/test
...

# grep -E '# (original|duplicate)' /tmp/out
original_cmd '/tmp/test/a' # original
cp_hardlink '/tmp/test/b' '/tmp/test/a' # duplicate

# rmlint -o summary -T duplicates -c sh:hardlink -g -o sh:/tmp/out /tmp/test
...

# grep -E '# (original|duplicate)' /tmp/out
original_cmd '/tmp/test/a' # original
remove_cmd '/tmp/test/b' '/tmp/test/a' # duplicate
```

Notice how in the second run, it uses `remove_cmd` instead of `cp_hardlink`.
This seems to be a combination of using `-o summary` and `-g` in that order. Removing either of these options causes it to work fine.

**Comment by sahib** [comment]
Hello @phemmer,

this surely is unfortunate and not very obvious... but not directly a bug. Let me explain:
The ``-g`` is your culprit here, since it clears all options and formatters set *before*:

```
-g --progress / -G --no-progress (default)
     Convenience shortcut for -o progressbar -o summary -o sh:rmlint.sh -VVV.
     Note: This flag clears all previous outputs.
           Specify any additional outputs after this flag!
```

So as a workaround you can specify the ``-g`` as first argument and it ``cp_hardlink`` will be emitted to the shell script (it would be actually enough to move the ``-g`` before the ``-c sh:hardlink``).

So far  the explanation, but I guess this needs fixing. Easiest solution is likely to not clear options or make ``-g`` only clear output if only the default outputs are set (which was the original intention).

**Comment by sahib** [comment]
Sorry for the very late answer, but finally remembered this issue.
It should be fixed by 8dc87154. Now ``-g`` only removes all ``pretty`` formatters (which is the default one that prints out that list of ``ls`` and ``rm`` lines). Would be nice if you could confirm it works for you.
